lib/liquid/context.rb

```diff
@@ -223,6 +204,13 @@ def lookup_and_evaluate(obj, key)
 
     private
 
+    def internal_error
+      # raise and catch to set backtrace and cause on exception
```

Comment:
Is this because `return Liquid::InternaError.new("internal")` would not have a backtrace? Maybe something like `#set_backtrace(caller)` would be easier to understand? Also, shouldn't the backtrace of this exception be the same backtrace as the one of the original exception (`e` above)?